parsec/core/mountpoint/manager.py

```diff
@@ -43,28 +49,71 @@ def get_mountpoint_runner():
             return fuse_mountpoint_runner
 
 
-def get_default_mountpoint(device_id: DeviceID):
-    if os.name == "nt":
-        return Path("Z:")
-    else:
-        return Path(f"/media/{device_id}")
+async def disabled_runner(*args, **kwargs):
+    raise MountpointDisabled()
+
+
+class MountpointManager:
+    def __init__(self, base_mountpoint, fs, runner, nursery):
+        self._base_mountpoint = base_mountpoint
+        self._fs = fs
+        self._runner = runner
+        self._nursery = nursery
+        self._mountpoint_tasks = {}
+
+    async def mount_workspace(self, workspace: str):
+        if workspace in self._mountpoint_tasks:
+            raise MountpointAlreadyMounted(f"Workspace `{workspace}` already mounted.")
+
+        try:
+            await self._fs.stat(f"/{workspace}")
+        except FileNotFoundError as exc:
+            raise MountpointConfigurationError(f"Workspace `{workspace}` doesn't exist") from exc
+
+        mountpoint = self._base_mountpoint / f"{self._fs.device.user_id}-{workspace}"
+        runner_task = await start_task(self._nursery, self._runner, workspace, mountpoint)
+        self._mountpoint_tasks[workspace] = runner_task
+
+    async def unmount_workspace(self, workspace: str):
+        if workspace not in self._mountpoint_tasks:
+            raise MountpointNotMounted(f"Workspace `{workspace}` not mounted.")
+
+        await self._mountpoint_tasks[workspace].cancel_and_join()
+        del self._mountpoint_tasks[workspace]
+
+    async def mount_all(self):
+        stat = await self._fs.stat("/")
+        for workspace in stat["children"]:
+            try:
+                await self.mount_workspace(workspace)
+            except MountpointAlreadyMounted:
+                pass
+
+    async def unmount_all(self):
+        for workspace in list(self._mountpoint_tasks.keys()):
+            await self.unmount_workspace(workspace)
 
 
 @asynccontextmanager
-async def mountpoint_manager(fs, event_bus, mountpoint, nursery, *, debug: bool = False, **config):
+async def mountpoint_manager_factory(
```

Comment:
Why the renaming? Isn't `mountpoint_manager` explicit enough? Or `mountpoint_manager_context` since it is an async context?

Reply:
`factory` is to indicate this function is not the mountpoint manager (i.e. call this function won't start fuse operation for instance) but will provide an the mountpoint manager (which expose an API to start/stop mountpoints)

Reply:
I see your point, although I don't think `factory` carries much information as a name. In my opinion, the following line feels a bit weird:
```python
async with mountpoint_manager_factory(...) as mountpoint_manager:
```
because we actually mean:
```python
async with mountpoint_management_context(...) as mountpoint_manager:
```
Not a big deal though, it's mostly nitpicking :smile: